pkg/apis/hive/v1/agent/installstrategy.go

```diff
@@ -0,0 +1,84 @@
+package agent
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// InstallStrategy is the install strategy configuration for provisioning a cluster with the
+// Agent based assisted installer.
+type InstallStrategy struct {
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking Networking `json:"networking"`
+
+	// TODO: consider moving higher into ClusterDeployment in future?
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+
+	// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+	ProvisionRequirements ProvisionRequirements `json:"provisionRequirements"`
+}
+
+// ProvisionRequirements defines configuration for when the installation is ready to be launched automatically.
+type ProvisionRequirements struct {
+
+	// ControlPlaneAgents is the number of matching approved and ready Agents with the control plane role
+	// required to launch the install. Must be either 1 or 3.
+	ControlPlaneAgents int `json:"controlPlaneAgents"`
+
+	// WorkerAgents is the minimum number of matching approved and ready Agents with the worker role
+	// required to launch the install.
+	// +optional
+	WorkerAgents int `json:"workerAgents,omitempty"`
+
+	// AgentSelector is a label selector used for associating relevant custom resources with this cluster.
+	// (Agent, BareMetalHost, etc)
+	AgentSelector metav1.LabelSelector `json:"agentSelector"`
+}
```

Comment:
defining how to select from space and the constraint of defining provision complete in same struct seems overloaded. I think maybe we should move the selector out to Strategy?

Reply:
Good point, moving up a level.